packages/synapse-react-client/src/mocks/msw/handlers/teamHandlers.ts

```diff
@@ -61,12 +61,17 @@ export function getTeamListHandler(backendOrigin: string) {
     `${backendOrigin}/repo/v1/teamList`,
     async (req, res, ctx) => {
       const requestBody: { list: number[] } = await req.json()
-
-      const teams = requestBody.list
-        .map(teamId => {
-          return getMockTeamById(String(teamId))
-        })
-        .filter((team): team is Team => !!team)
+      const teams: Team[] = []
+      for (const teamId of requestBody.list) {
+        const team = getMockTeamById(teamId.toString())
+        if (!team) {
+          const errorResponse: SynapseApiResponse<ListWrapper<Team>> = {
+            reason: `Team with id ${teamId} not found`,
+          }
+          return res(ctx.status(404), ctx.json(errorResponse))
+        }
```

Comment:
Return an error response when a team is not found to replicate the backend's behavior